src/core/identity.cc

```diff
@@ -50,19 +50,26 @@
 namespace i2p {
 namespace data {
 
+// TODO(unassigned): identity implementation needs a big refactor
+
+// TODO(unassigned): keep an eye open for alignment issues and for hacks like:
+// copy public and signing keys together
+//memcpy(public_key, keys.public_key, sizeof(public_key) + sizeof(signing_key));
+
 Identity& Identity::operator=(const Keys& keys) {
-  // copy public and signing keys together
-  memcpy(public_key, keys.public_key, sizeof(public_key) + sizeof(signing_key));
+  memcpy(public_key, keys.public_key, sizeof(public_key));
   memset(&certificate, 0, sizeof(certificate));
   return *this;
 }
 
-size_t Identity::FromBuffer(
+// TODO(unassigned): unused, remove after refactor
```

Comment:
I want to leave the implementation until **after** the refactor. I hate deleting code that *may* be even the slightest useful in the future. If not, we can clobber it then when we write a cleaner impl.